File: algorithms/cpp/maximum69Number/maximum69Number.cpp
// Source: https://leetcode.com/problems/maximum-69-number/
// Author: Miao Zhang
// Date:   2021-04-23

class Solution {
public:
    int maximum69Number (int num) {
        string s = to_string(num);
        for (int i = 0; i < s.length(); i++) {
            if (s[i] == '6') {
                s[i] = '9';
                break;
            }
        }
        return stoi(s);
    }
};

File: algorithms/python/maximum69Number/maximum69Number.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-

# Source: https://leetcode.com/problems/maximum-69-number/
# Author: Miao Zhang
# Date:   2021-04-23

class Solution:
    def maximum69Number (self, num: int) -> int:
        return int(str(num).replace('6', '9', 1))
